modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm

```diff
@@ -23,68 +23,242 @@
     "\"What the fuck does any of this mean?\" - Doctor Micheals, reading their CentCom report about the new 'hires'.")
 
+
 /datum/species/hemophage/prepare_human_for_preview(mob/living/carbon/human/human)
 	human.skin_tone = "albino"
 	human.hair_color = "#1d1d1d"
 	human.hairstyle = "Pompadour (Big)"
 	human.update_mutant_bodyparts(TRUE)
 	human.update_body(TRUE)
+
+
+/obj/item/organ/internal/tongue/hemophage
+	name = "hemophage tongue"
+	color = "#333333"
+	actions_types = list(/datum/action/item_action/organ_action/hemophage)
+	COOLDOWN_DECLARE(drain_cooldown)
+
+
+/datum/action/item_action/organ_action/hemophage
+	name = "Drain Victim"
+	desc = "Leech blood from any carbon victim you are passively grabbing."
+
+/// Maximum an Hemophage will drain, they will drain less if they hit their cap.
+#define HEMOPHAGE_DRAIN_AMOUNT 50
+
+
+/datum/action/item_action/organ_action/hemophage/Trigger(trigger_flags)
+	. = ..()
+	if(!iscarbon(owner))
+		return
+
+	var/mob/living/carbon/hemophage = owner
+	var/obj/item/organ/internal/tongue/hemophage/tongue = target
+
+	if(!COOLDOWN_FINISHED(tongue, drain_cooldown))
+		hemophage.balloon_alert(hemophage, "just drained blood, wait a few seconds!")
+		return
+
+	if(!hemophage.pulling || !iscarbon(hemophage.pulling))
+		return
+
+	var/mob/living/carbon/victim = hemophage.pulling
+	if(hemophage.blood_volume >= BLOOD_VOLUME_MAXIMUM)
+		hemophage.balloon_alert(hemophage, "already full!")
+		return
+
+	if(victim.stat == DEAD)
+		hemophage.balloon_alert(hemophage, "needs a living victim!")
+		return
+
+	if(!victim.blood_volume || (victim.dna && ((NOBLOOD in victim.dna.species.species_traits) || victim.dna.species.exotic_blood)))
+		hemophage.balloon_alert(hemophage, "[victim] doesn't have blood!")
+		return
+
+
+	var/blood_volume_difference = BLOOD_VOLUME_MAXIMUM - hemophage.blood_volume //How much capacity we have left to absorb blood
+	// We start by checking that the victim is a human and they have a client, so we can give them the
+	// beneficial status effect for drinking higher-quality blood.
+	var/is_target_human_with_client = istype(victim, /mob/living/carbon/human) && victim.client
+
+	if(ismonkey(victim))
+		if(hemophage.blood_volume >= BLOOD_VOLUME_NORMAL)
+			hemophage.balloon_alert(hemophage, "their inferior blood cannot sate you any further!")
+			return
+
+		is_target_human_with_client = FALSE // Sorry, not going to get the status effect from monkeys, even if they have a client in them.
+
+		blood_volume_difference = BLOOD_VOLUME_NORMAL - hemophage.blood_volume
+
+
+	COOLDOWN_START(tongue, drain_cooldown, 3 SECONDS)
+	if(victim.can_block_magic(MAGIC_RESISTANCE_HOLY, charge_cost = 0))
+		victim.show_message(span_warning("[hemophage] tries to bite you, but stops before touching you!"))
+		to_chat(hemophage, span_warning("[victim] is blessed! You stop just in time to avoid catching fire."))
```

Comment:
why are you using to_chat here and below when you were using balloon alerts

Reply:
Because this is the kind of thing that you wouldn't want to miss as it happens.